tests/components/counter/common.py

```diff
@@ -19,7 +19,7 @@
 @bind_hass
 def async_increment(hass, entity_id):
     """Increment a counter."""
-    hass.async_add_job(
+    hass.create_task(
```

Comment:
We should use the async api (`hass.async_create_task`) inside the event loop.

Reply:
Apparently I really needed that sleep last night 🤦 

https://github.com/home-assistant/core/pull/113441

Reply:
Thanks for catching that